# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I chose the **Content Refresh Prioritization** lane because organizations have many webpages but limited time and resources to update them. I want to use machine learning to identify which pages are most likely to lose search performance so they can be refreshed first. This project focuses on supporting better decisions using data instead of relying only on manual judgment.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Research Question:** Which webpages should be prioritized for content refresh to reduce the risk of declining search performance?

**Decision:** Decide which pages should be refreshed first.

**Who takes the action:** SEO managers or content teams.

**Action:** Update the highest-priority pages before they lose significant search traffic.

**Cost of a wrong recommendation:** If the wrong pages are refreshed, time and resources are wasted while important pages continue losing traffic and visibility.

**Why ML helps:** Machine learning can combine multiple signals and identify patterns that are difficult to detect using simple rules.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nTrend Direction")
print(df["trend_direction"].value_counts())

print("\nAverage CTR")
print(df["ctr"].mean())

print("\nAverage Search Volume")
print(df["search_volume"].mean())

Rows: 30000
Columns: 44

Trend Direction
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR
0.5107333333333334

Average Search Volume
158.88239139909922


### Observations

- The dataset contains **30000** rows.
- The average CTR is **0.510733**.
- The average search volume is **158.882391**.

These numbers show that the dataset is large enough to investigate which pages should be refreshed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project provides decision support based on observed patterns in the available data.
It does not prove causation or predict Google's ranking algorithm. Any recommendations should be validated before being used in real-world decision making.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.